app/models/spree/credit_card.rb

```diff
@@ -79,11 +79,17 @@ def display_number
     end
 
     def actions
-      %w{capture void credit}
+      %w{capture void credit resend_authorization_email}
+    end
+
+    def can_resend_authorization_email?(payment)
+      payment.pending? && payment.cvv_response_message.present?
```

Comment:
this :nose: like belongs to `payment` if it needs to ask it all to it. Although, I don't see this method used anywhere? Is there metaprogramming I'm not seeing :thinking: ?

Reply:
Yep, metaprogramming. It's on [line 113 of payment.rb](https://github.com/openfoodfoundation/openfoodnetwork/blob/59c981e14d77c20e6d409df3006aa39e4f6d736d/app/models/spree/payment.rb#L113).

Also agree that it might be good to move these methods to `payment`... but it might get tricky because of that metaprogramming. Let me think about it a bit more.

Reply:
ah, I see, thanks.

Reply:
I think that keeping these methods in credit_card.rb (as opposed to moving it to payment.rb) is acceptable; it allows for the `payment_source` to be something other than a credit card. 

I'm going to mark as resolved but feel free to unresolve if you want to discuss more.